icu4c/source/common/unicode/platform.h

```diff
@@ -168,7 +168,7 @@
 #   define U_PLATFORM U_PF_LINUX
 #elif defined(__APPLE__) && defined(__MACH__)
 #   include <TargetConditionals.h>
-#   if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE  /* variant of TARGET_OS_MAC */
+#   if (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE) && (defined(TARGET_OS_MACCATALYST) && !TARGET_OS_MACCATALYST)   /* variant of TARGET_OS_MAC */
```

Comment:
No, my intention is `U_PF_DARWIN` to be set when building for maccatalyst since that slice has to match with the macOS slice and maccatalyst triple also sets  `TARGET_OS_IPHONE`. 
Its a little confusing but `TARGET_OS_IPHONE != TARGET_OS_IOS`. 

MacCatalyst basically means the same macOS binary can run on macOS and iOS but there are specific api differences already set in ICU. Internally this change exists and hasn't caused issues so those differences may not be exposed to clients. The other option is to make MacOS == iOS == MacCatalyst so there is no need specify Macro distinctions.